src/v/redpanda/admin_server.cc

```diff
@@ -1484,86 +1483,90 @@ static bool match_scram_credential(
     }
 }
 
-void admin_server::register_security_routes() {
-    register_route<superuser>(
-      ss::httpd::security_json::create_user,
-      [this](std::unique_ptr<ss::httpd::request> req)
-        -> ss::future<ss::json::json_return_type> {
-          auto doc = parse_json_body(*req);
+ss::future<ss::json::json_return_type>
+admin_server::create_user_handler(std::unique_ptr<ss::httpd::request> req) {
+    auto doc = parse_json_body(*req);
 
-          auto credential = parse_scram_credential(doc);
+    auto credential = parse_scram_credential(doc);
 
-          if (!doc.HasMember("username") || !doc["username"].IsString()) {
-              throw ss::httpd::bad_request_exception(
-                fmt::format("String username missing"));
-          }
+    if (!doc.HasMember("username") || !doc["username"].IsString()) {
+        throw ss::httpd::bad_request_exception(
+          fmt::format("String username missing"));
```

Comment:
These throws are fine, but I'd like to point out `ss::coroutine::exception`, but I'm not convinced it's the easiest thing to use, it has some API quirks.